docs/programming_guide/experiment_tracking.rst

```diff
@@ -0,0 +1,131 @@
+.. _experiment_tracking:
+
+###################
+Experiment Tracking
+###################
+
+***********************
+Overview and Approaches
+***********************
+
+In a federated computing setting, the data is distributed across multiple devices or systems, and training is run
+on each device independently while preserving each client's data privacy.
+
+Assuming a federated system consisting of one server and many clients and the server coordinating the ML training of clients,
+we can interact with ML experiment tracking tools in two different ways: 
+
+    - Client-side experiment tracking: Each client will directly send the log metrics/parameters to the ML experiment tracking server or local file system
```

Comment:
Can we add some examples here like:

the ML experiment tracking server (like MLFlow) or local file system (like tensorboard)